### PR TITLE
support mute alert

### PR DESCRIPTION
**Pull request by xudifsd**
Add muting alert facilities, rely on amtool to manage alert. Provide a script for generating amtool config.

@scarlett2018 please review the doc.

### CODE REVIEW

File: utilities/gen-amtool-config.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    amtool_dir = os.path.join(home, ".config/amtool")
+    if not os.path.exists(amtool_dir):
+        os.makedirs(amtool_dir)
+    config = os.path.join(amtool_dir, "config.yml")
+    if os.path.isfile(config) and not args.force:
+        sys.stderr.write("{} already exist, specify -f to overwirte\n".format(config))
+        sys.exit(1)
+
+    with codecs.open(config, "w", "utf-8") as f:
+        f.write(amtool_config.format(host,port))
```
[the remark this answers is not in the document]
Comment (xudifsd) [reply utilities/gen-amtool-config.py]: [here](https://github.com/Microsoft/pai/pull/1261/files/870a5a000343a65fad47a65fc70479f445a81cac#diff-425c9b0b62750fcb61f40dd6e330837fR30)

File: pai-management/Dockerfile [hunk 2]
```diff
@@ -52,20 +53,33 @@ RUN apt-get -y update && \
...
 RUN wget https://download.docker.com/linux/static/stable/x86_64/docker-17.06.2-ce.tgz
-RUN cp docker-17.06.2-ce.tgz /usr/local
-RUN tar xzvf /usr/local/docker-17.06.2-ce.tgz
-RUN cp -r /docker/* /usr/bin/
+RUN tar xzvf docker-17.06.2-ce.tgz
+RUN mv docker/* /usr/local/bin/
+
+# alert manager tool
+RUN wget https://github.com/prometheus/alertmanager/releases/download/v0.15.2/alertmanager-0.15.2.linux-amd64.tar.gz
+RUN tar xzvf alertmanager-0.15.2.linux-amd64.tar.gz
```
Comment (YanjieGao) [inline pai-management/Dockerfile]: Alertmanager is launched at dev-box container? Why should make it at this container

Comment (xudifsd) [reply pai-management/Dockerfile]: this is not for launch alert manager in dev-box, it's just for download amtool which is a client that can mute alert.

Comment (YanjieGao) [reply pai-management/Dockerfile]: ok

File: prometheus/doc/alert-manager.md [hunk 3]
```diff
@@ -58,3 +58,28 @@ cd $pai-management
 ```
 
 Please fire a pull request if you find any rule useful.
+
+# Muting firing alert
+
+OpenPai leverage [amtool](https://github.com/prometheus/alertmanager#amtool) to interactive with
```
Comment (YanjieGao) [inline prometheus/doc/alert-manager.md]: OpenPAI syntax

Comment (xudifsd) [reply prometheus/doc/alert-manager.md]: changed

File: prometheus/doc/alert-manager.md [hunk 4]
```diff
@@ -58,3 +58,28 @@ cd $pai-management
...
+OpenPAI leverage [amtool](https://github.com/prometheus/alertmanager#amtool) to interactive with
+alert manager. If you're using [dev-box](../../pai-management/doc/cluster-bootup.md#c-step-0) to do
+cluster management, the amtool might has already been installed for you, you can try to execute
+`amtool` in your shell to test. If it hasn't been installed, please install it using method provided
+by amtool [README](https://github.com/prometheus/alertmanager#install-1).
+
+To ease configuration, OpenPAI provides a script that generate amtool config file for you, you can use
+
+```
+python utilities/gen-amtool-config.py -p /cluster-configuration
```
Comment (YanjieGao) [inline prometheus/doc/alert-manager.md]: will this trigger muting ? or just reconfig? if trigger, please tell user after execute will mute

Comment (xudifsd) [reply prometheus/doc/alert-manager.md]: this only generate configuration for amtool, user should call amtool for muting or something.

Comment (xudifsd) [reply prometheus/doc/alert-manager.md]: I mentioned that [here](https://github.com/Microsoft/pai/pull/1261/files/6e754af8b9c788256a740872feb3f101e6276908#diff-f27c2b85c290aaf4f5f812da80fad884R84)

Comment (YanjieGao) [reply prometheus/doc/alert-manager.md]: for me first time read this, not aware and have the question above.

copy this (or tell user could find Silence an alert cmd at that file will be intuitive): 
Silence an alert

$ amtool silence add alertname=Test_Alert
b3ede22e-ca14-4aa0-932c-ca2f3445f926

$ amtool silence add alertname="Test_Alert" instance=~".+0"
e48cb58a-0b17-49ba-b734-3585139b1d25

Comment (xudifsd) [reply prometheus/doc/alert-manager.md]: updaed

**Pull request merged**